src/main/webapp/app/exercises/programming/manage/grading/testcase-analysis/testcase-analysis.component.ts

```diff
@@ -0,0 +1,87 @@
+import { Component, Input, OnInit } from '@angular/core';
+import { ArtemisSharedModule } from 'app/shared/shared.module';
+import { FeedbackDetailsWithResultIdsDTO, SimplifiedTask, TestcaseAnalysisService } from 'app/exercises/programming/manage/grading/testcase-analysis/testcase-analysis.service';
+import { Observable } from 'rxjs';
+import { concatMap, tap } from 'rxjs/operators';
+import { HttpResponse } from '@angular/common/http';
+
+type FeedbackDetail = {
+    count: number;
+    detailText: string;
+    testcase: string;
+    task: number;
+};
+
+@Component({
+    selector: 'jhi-testcase-analysis',
+    templateUrl: './testcase-analysis.component.html',
+    standalone: true,
+    imports: [ArtemisSharedModule],
+    providers: [TestcaseAnalysisService],
+})
+export class TestcaseAnalysisComponent implements OnInit {
+    @Input() exerciseTitle?: string;
+    @Input() exerciseId?: number;
+    resultIds: number[] = [];
+    tasks: SimplifiedTask[] = [];
+    feedback: FeedbackDetail[] = [];
+
+    constructor(private simplifiedProgrammingExerciseTaskService: TestcaseAnalysisService) {}
+
+    ngOnInit(): void {
+        if (this.exerciseId) {
+            this.loadTasks(this.exerciseId)
+                .pipe(concatMap(() => this.loadFeedbackDetails(this.exerciseId!)))
+                .subscribe();
+        }
+    }
+
+    loadTasks(exerciseId: number): Observable<SimplifiedTask[]> {
+        return this.simplifiedProgrammingExerciseTaskService.getSimplifiedTasks(exerciseId).pipe(
+            tap((tasks) => {
+                this.tasks = tasks;
+            }),
+        );
+    }
+
+    loadFeedbackDetails(exerciseId: number): Observable<HttpResponse<FeedbackDetailsWithResultIdsDTO>> {
+        return this.simplifiedProgrammingExerciseTaskService.getFeedbackDetailsForExercise(exerciseId).pipe(
+            tap((response) => {
+                this.resultIds = response.body?.resultIds || [];
+                const feedbackArray = response.body?.feedbackDetails || [];
+                this.saveFeedback(feedbackArray);
+            }),
+        );
+    }
+
+    saveFeedback(feedbackArray: { detailText: string; testCaseName: string }[]): void {
```

Comment:
feedbackArray --> `feedbacks` or `feedbackDetails`

and you could also use the interface